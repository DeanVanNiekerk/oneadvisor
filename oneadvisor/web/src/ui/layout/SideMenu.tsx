import { Icon, Layout, Menu as MenuAD } from "antd";
import React, { Component, CSSProperties } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import { hasPermissionsMenuGroup, hasRoles, hasUseCases } from "@/app/identity";
import { defaultOpenGroupNames } from "@/config/menu";
import { roleSelector, useCaseSelector } from "@/state/auth";
import { currentApplicationSelector, currentMenuSelector } from "@/state/context/selectors";
import { Application, Menu, MenuLink } from "@/state/context/types";
import { RootState } from "@/state/rootReducer";

const { SubMenu, Item } = MenuAD;
const { Sider } = Layout;

type Props = {
    menu: Menu;
    application: Application;
    useCases: string[];
    roles: string[];
};

type State = {
    collapsed: boolean;
};

class SideMenu extends Component<Props, State> {
    constructor(props) {
        super(props);

        this.state = {
            collapsed: false,
        };
    }

    onCollapse = collapsed => {
        this.setState({ collapsed });
    };

    getMenuItemStyle = (link: MenuLink, application: Application): CSSProperties => {
        if (!link.isCurrent) return {};
        return {
            backgroundColor: `${application.color}`,
        };
    };

    render() {
        return (
            <Sider
                collapsible
                collapsed={this.state.collapsed}
                onCollapse={this.onCollapse}
                width={225}
                style={{ background: "#fff" }}
            >
                <MenuAD
                    theme="dark"
                    mode="inline"
                    defaultOpenKeys={defaultOpenGroupNames()}
                    style={{ height: "100%", borderRight: 0 }}
                    selectedKeys={[]}
                >
                    {this.props.menu.groups
                        .filter(group =>
                            hasPermissionsMenuGroup(group, this.props.useCases, this.props.roles)
                        )
                        .map(group => (
                            <SubMenu
                                key={group.name}
                                title={
                                    <span>
                                        <Icon type={group.icon} />
                                        <span>{group.name}</span>
                                    </span>
                                }
                            >
                                {group.links
                                    .filter(
                                        link =>
                                            hasUseCases(link.useCases, this.props.useCases) &&
                                            hasRoles(link.roles, this.props.roles)
                                    )
                                    .map(link => (
                                        <Item
                                            key={link.relativePath}
                                            style={this.getMenuItemStyle(
                                                link,
                                                this.props.application
                                            )}
                                        >
                                            <Link
                                                to={`${this.props.menu.relativePath}${link.relativePath}`}
                                            >
                                                <span>{link.name}</span>
                                            </Link>
                                        </Item>
                                    ))}
                            </SubMenu>
                        ))}
                </MenuAD>
            </Sider>
        );
    }
}

const mapStateToProps = (state: RootState) => {
    return {
        menu: currentMenuSelector(state),
        application: currentApplicationSelector(state),
        useCases: useCaseSelector(state),
        roles: roleSelector(state),
    };
};

export default connect(mapStateToProps)(SideMenu);
