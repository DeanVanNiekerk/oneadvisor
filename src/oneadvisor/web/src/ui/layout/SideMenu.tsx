import { Layout, Menu as MenuAD } from "antd";
import React, { Component, CSSProperties } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import { hasPermissionsMenuGroup, hasRoles, hasUseCases } from "@/app/identity";
import { defaultOpenGroupNames } from "@/config/menu";
import { RootState } from "@/state";
import { roleSelector, useCaseSelector } from "@/state/auth";
import { currentMenuSelector, currentRootNavigationItemSelector } from "@/state/context/selectors";
import { Menu, MenuLink, RootNavigationItem } from "@/state/context/types";
import { Icon } from "@/ui/controls/common/Icon";

const { SubMenu, Item } = MenuAD;
const { Sider } = Layout;

type Props = {
    menu: Menu;
    rootNavigationItem: RootNavigationItem;
    useCases: string[];
    roles: string[];
};

type State = {
    collapsed: boolean;
};

class SideMenu extends Component<Props, State> {
    constructor(props) {
        super(props);

        this.state = {
            collapsed: false,
        };
    }

    onCollapse = (collapsed) => {
        this.setState({ collapsed });
    };

    getMenuItemStyle = (link: MenuLink, item: RootNavigationItem): CSSProperties => {
        if (!link.isCurrent) return {};
        return {
            backgroundColor: `${item.color}`,
        };
    };

    render() {
        return (
            <Sider
                collapsible
                collapsed={this.state.collapsed}
                onCollapse={this.onCollapse}
                width={225}
                style={{ background: "#fff" }}
            >
                <MenuAD
                    theme="dark"
                    mode="inline"
                    defaultOpenKeys={defaultOpenGroupNames()}
                    style={{ height: "100%", borderRight: 0 }}
                    selectedKeys={[]}
                >
                    {this.props.menu.groups
                        .filter((group) =>
                            hasPermissionsMenuGroup(group, this.props.useCases, this.props.roles)
                        )
                        .map((group) => (
                            <SubMenu
                                key={group.name}
                                title={
                                    <span>
                                        <Icon name={group.icon} />
                                        <span>{group.name}</span>
                                    </span>
                                }
                            >
                                {group.links
                                    .filter(
                                        (link) =>
                                            hasUseCases(link.useCases, this.props.useCases) &&
                                            hasRoles(link.roles, this.props.roles)
                                    )
                                    .map((link) => (
                                        <Item
                                            key={link.relativePath}
                                            style={this.getMenuItemStyle(
                                                link,
                                                this.props.rootNavigationItem
                                            )}
                                        >
                                            <Link
                                                to={`${this.props.menu.relativePath}${link.relativePath}`}
                                            >
                                                <span>{link.name}</span>
                                            </Link>
                                        </Item>
                                    ))}
                            </SubMenu>
                        ))}
                </MenuAD>
            </Sider>
        );
    }
}

const mapStateToProps = (state: RootState) => {
    return {
        menu: currentMenuSelector(state),
        rootNavigationItem: currentRootNavigationItemSelector(state),
        useCases: useCaseSelector(state),
        roles: roleSelector(state),
    };
};

export default connect(mapStateToProps)(SideMenu);
