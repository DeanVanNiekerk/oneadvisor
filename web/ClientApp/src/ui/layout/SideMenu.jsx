import React from 'react';
import styled from 'styled-components'
import { Link } from 'react-router-dom';
import { Nav, NavItem } from 'reactstrap';

const Wrapper = styled.div`
    order: 0;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    border-right: 1px solid rgba(0, 0, 0, 0.1);
    position: sticky !important;
    top: 3.5rem;
    z-index: 1000;
    height: calc(100vh - 3.5rem);
    overflow-y: auto;
`;

class SideMenu extends React.Component {

    render() {

        return (
            <Wrapper {...this.props}>
               
                <Nav className="flex-column">
                    <NavItem>
                        <Link className="nav-link active" to="/directory/users">Users</Link>
                    </NavItem>
                    <NavItem>
                        <Link className="nav-link active" to="/directory/users">Organisations</Link>
                    </NavItem>
                </Nav>

            </Wrapper>
        );
    }
}

export default SideMenu



