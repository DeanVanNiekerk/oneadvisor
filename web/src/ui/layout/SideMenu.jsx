// @flow

import React, { Component } from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Layout, Menu as MenuAD, Icon } from 'antd';

import config from '@/config/config';
import type { State as RootState } from '@/state/rootReducer';
import type { Menu, MenuLink, Application } from '@/state/context/types';
import {
    currentMenuSelector,
    currentApplicationSelector
} from '@/state/context/selectors';

const { SubMenu, Item } = MenuAD;
const { Sider } = Layout;

const MenuItem = styled(Item)`
    ${props =>
        props.link.isCurrent &&
        css`
            background-color: ${props => props.application.color} !important;
        `}
`;

type Props = {
    className: string,
    menu: Menu,
    application: Application
};

class SideMenu extends Component<Props> {
    render() {
        return (
            <Sider width={225} style={{ background: '#fff' }}>
                <MenuAD
                    theme="dark"
                    mode="inline"
                    defaultOpenKeys={this.props.menu.groups.map(g => g.name)}
                    style={{ height: '100%', borderRight: 0 }}
                >
                    {this.props.menu.groups.map(group => (
                        <SubMenu
                            key={group.name}
                            title={<span>{group.name}</span>}
                        >
                            {group.links.map(link => (
                                <MenuItem
                                    key={link.relativePath}
                                    link={link}
                                    application={this.props.application}
                                >
                                    <Link
                                        to={`${this.props.menu.relativePath}${link.relativePath}`}
                                    >
                                        <span>
                                            <Icon type={link.icon} />
                                            <span>{link.name}</span>
                                        </span>
                                    </Link>
                                </MenuItem>
                            ))}
                        </SubMenu>
                    ))}
                </MenuAD>
            </Sider>
          
        );
    }
}

const mapStateToProps = (state: RootState) => ({
    menu: currentMenuSelector(state),
    application: currentApplicationSelector(state)
});

export default connect(mapStateToProps)(SideMenu);
