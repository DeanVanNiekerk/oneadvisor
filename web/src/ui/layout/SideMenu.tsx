import { Icon, Layout, Menu as MenuAD } from 'antd';
import React, { Component, CSSProperties } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { hasUseCases, hasUseCasesMenuGroup } from '@/app/identity';
import { allGroupNames } from '@/config/menu';
import { useCaseSelector } from '@/state/auth';
import { currentApplicationSelector, currentMenuSelector } from '@/state/context/selectors';
import { Application, Menu, MenuLink } from '@/state/context/types';
import { RootState } from '@/state/rootReducer';

const { SubMenu, Item } = MenuAD;
const { Sider } = Layout;

type Props = {
    menu: Menu;
    application: Application;
    useCases: string[];
};

class SideMenu extends Component<Props> {
    getMenuItemStyle = (
        link: MenuLink,
        application: Application
    ): CSSProperties => {
        if (!link.isCurrent) return {};
        return {
            backgroundColor: `${application.color}`,
        };
    };

    render() {
        return (
            <Sider width={225} style={{ background: "#fff" }}>
                <MenuAD
                    theme="dark"
                    mode="inline"
                    defaultOpenKeys={allGroupNames()}
                    style={{ height: "100%", borderRight: 0 }}
                    selectedKeys={[]}
                >
                    {this.props.menu.groups
                        .filter(group =>
                            hasUseCasesMenuGroup(group, this.props.useCases)
                        )
                        .map(group => (
                            <SubMenu
                                key={group.name}
                                title={<span>{group.name}</span>}
                            >
                                {group.links
                                    .filter(link =>
                                        hasUseCases(
                                            link.useCases,
                                            this.props.useCases
                                        )
                                    )
                                    .map(link => (
                                        <Item
                                            key={link.relativePath}
                                            style={this.getMenuItemStyle(
                                                link,
                                                this.props.application
                                            )}
                                        >
                                            <Link
                                                to={`${
                                                    this.props.menu.relativePath
                                                }${link.relativePath}`}
                                            >
                                                <span>
                                                    <Icon type={link.icon} />
                                                    <span>{link.name}</span>
                                                </span>
                                            </Link>
                                        </Item>
                                    ))}
                            </SubMenu>
                        ))}
                </MenuAD>
            </Sider>
        );
    }
}

const mapStateToProps = (state: RootState) => {
    return {
        menu: currentMenuSelector(state),
        application: currentApplicationSelector(state),
        useCases: useCaseSelector(state),
    };
};

export default connect(mapStateToProps)(SideMenu);
