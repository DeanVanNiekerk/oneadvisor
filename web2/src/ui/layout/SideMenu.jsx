// @flow

import React, { Component } from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import Icon from "@material-ui/core/Icon";
import DrawerMUI from "@material-ui/core/Drawer";
import Toolbar from "@material-ui/core/Toolbar";
import List from "@material-ui/core/List";
import Divider from "@material-ui/core/Divider";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import AppBar from "@material-ui/core/AppBar";
import ListSubheader from "@material-ui/core/ListSubheader";

import type { State as RootState } from "state/rootReducer";
import type { Menu, Application } from "state/context/types";
import {
  currentMenuSelector,
  currentApplicationSelector
} from "state/context/selectors";

// const Wrapper = styled.div`
//     order: 0;
//     border-right: 3px solid #d4d4d4;
//     background-color: #f0f0f0;
//     position: sticky !important;
//     top: 3.5rem;
//     z-index: 1000;
//     height: calc(100vh - 3.7rem);
//     overflow-y: auto;
//     padding-top: 1rem;

//     font-size: 0.9rem;

//     -webkit-box-shadow: inset -5px 0px 5px -4px rgba(212,212,212,1);
//     -moz-box-shadow: inset -5px 0px 5px -4px rgba(212,212,212,1);
//     box-shadow: inset -5px 0px 5px -4px rgba(212,212,212,1);
// `;

// const NavItem = styled(NavItemBs)`
//     ${props => props.link.isCurrent && css`
//         background-color: #d4d4d4;
//         border-left: 6px solid ${props => props.application.color};
//     `}
//     ${props => !props.link.isCurrent && css`
//         border-left: 6px solid #f0f0f0;
//     `}
// `;

const drawerWidth = 240;

const Drawer = styled(DrawerMUI)`
  width: ${drawerWidth}px !important;
  flex-shrink: 0 !important;
`;
const styles = () => ({
  drawerPaper: {
    width: drawerWidth
  }
});

type Props = {
  className: string,
  menu: Menu,
  application: Application,
  classes: Object
};

class SideMenu extends Component<Props> {
  render() {
    return (
      <Drawer
        variant="permanent"
        classes={{
          paper: this.props.classes.drawerPaper
        }}
      >
        <AppBar position="static">
          <Toolbar />
        </AppBar>

        {this.props.menu.groups.map(group => {
          return (
            <List
              key={group.name}
              subheader={<ListSubheader>{group.name}</ListSubheader>}
            >
              {group.links.map(link => {
                return (
                  <ListItem
                    key={link.relativePath}
                    application={this.props.application}
                    link={link}
                  >
                    <ListItemIcon>
                      <Icon>{link.icon}</Icon>
                    </ListItemIcon>
                    <ListItemText
                      to={`${this.props.menu.relativePath}${link.relativePath}`}
                    >
                      {link.name}
                    </ListItemText>
                  </ListItem>
                );
              })}
            </List>
          );
        })}
      </Drawer>
      // <Wrapper>

      //     {this.props.menu.groups.map(group => {

      //         return (<div key={group.name}>
      //             <div className="pl-2 pb-1 text-uppercase">{group.name}</div>
      //             <Nav className="flex-column">

      //                 {group.links.map(link => {
      //                     return (
      //                         <NavItem key={link.relativePath} className="pl-3" application={this.props.application} link={link}>
      //                             <Icon className="float-left mt-2 text-dark">{link.icon}</Icon>
      //                             <Link className="float-left nav-link text-dark pl-2" to={`${this.props.menu.relativePath}${link.relativePath}`}>
      //                                 {link.name}
      //                             </Link>
      //                         </NavItem>
      //                     )})
      //                 }

      //             </Nav>
      //         </div>
      //         )
      //     })}

      // </Wrapper>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  menu: currentMenuSelector(state),
  application: currentApplicationSelector(state)
});

export default connect(mapStateToProps)(withStyles(styles)(SideMenu));
