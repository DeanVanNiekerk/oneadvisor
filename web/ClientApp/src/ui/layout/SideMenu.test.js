import React from 'react';
import SideMenu from './SideMenu';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom' 

describe('SideMenu', () => {

    it('renders correctly', () => {

        const tree = renderer
            .create(<Router><SideMenu /></Router>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});