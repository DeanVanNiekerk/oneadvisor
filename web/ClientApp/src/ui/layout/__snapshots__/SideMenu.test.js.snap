// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders correctly 1`] = `
<div
  className="sc-bdVaJa eixjvS"
>
  <ul
    className="flex-column nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/directory/users"
        onClick={[Function]}
      >
        Users
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/directory/users"
        onClick={[Function]}
      >
        Organisations
      </a>
    </li>
  </ul>
</div>
`;
